nix/services/loki.nix

```diff
@@ -0,0 +1,126 @@
+{ pkgs
+, lib
+, name
+, config
+, ...
+}:
+let
+  inherit (lib) types;
+  yamlFormat = pkgs.formats.yaml { };
+in
+{
+  options = {
+    description = ''
+      Configure loki.
+    '';
+    package = lib.mkPackageOption pkgs "grafana-loki" { };
+
+    httpAddress = lib.mkOption {
+      type = types.str;
+      description = "Which address to access loki from.";
+      default = "localhost";
+    };
+
+    httpPort = lib.mkOption {
+      type = types.int;
+      description = "Which HTTP port to loki should listen on.";
+      default = 3100;
+    };
+
+    grpcPort = lib.mkOption {
+      type = types.int;
+      description = "Which gRPC port to run loki should listen on.";
+      default = 9096;
+    };
+
+    extraConfig = lib.mkOption {
+      type = yamlFormat.type;
+      default = { };
+      description = lib.mdDoc ''
+        Specify the configuration for Loki in Nix.
+
+        See https://grafana.com/docs/loki/latest/configuration/ for available options.
+      '';
+    };
+
+    extraFlags = lib.mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      example = lib.literalExpression ''
+        [ "-config.expand-env=true" ]
+      '';
+      description = lib.mdDoc ''
```

Comment:
`lib.mdDoc` is a no-op.

See, for example, https://github.com/NixOS/nixpkgs/pull/303841